docs/grid_exchange_formats/going_further/datasources.md

```diff
@@ -0,0 +1,158 @@
+- A Zip archive is also already compressed so the compression format for `ZipArchiveDataSource` is always ZIP.
+- A Tar archive can be compressed by: BZIP2, GZIP, XZ or ZSTD. It can also not be compressed.
+
+Just like `DirectoryDataSource`, the archive datasources integrate the notions of base name and data extension. If not
+given as a parameter in the datasource constructor, the archive file name is even defined using the base name and the
+data extension, as `<directory>/<basename>.<dataExtension>.<archiveExtension>.<compressionExtension>` with the 
+compression extension being optional depending on the archive format. For example `network.xiidm.zip` contains
+`network.xiidm`.
+
+
+## Example
+
+Let's consider a directory containing the following files:
+
+```
+directory              
+├── network              
+├── network.south              
+├── network.xiidm.gz    
+├── network.v3.xiidm.gz
+├── network_mapping.csv.gz
+├── network.gz         
+└── toto.xiidm.gz  
+```
+
+A datasource on this directory could be used this way:
+
+```java
+// Creation of a directory datasource with compression
+GzDirectoryDataSource datasource = new GzDirectoryDataSource(testDir, "network", "xiidm", observer);
+
+// Check if some files exist in the datasource by using the `exists(String fileName)` method
+// Since the datasource uses Gzip compression, ".gz" is added to the provided fileName parameter
+datasource.exists("test.toto") // Returns false: the file "test.toto.gz" does not exist in the directory
+datasource.exists("network.south") // Returns false: the file "network.south.gz" does not exist
+datasource.exists("network.xiidm") // Returns true: the file "network.xiidm.gz" exists
+
+// Check if some files exist in the datasource by using the `exists(String fileName)` method
+datasource.exists("_south", "reduced") // Returns false: the file "network_south.reduced.gz" does not exist in the directory
+datasource.exists(null, "xiidm") // Returns true: the file "network.xiidm.gz" exists in the directory
+datasource.exists("_mapping", "csv") // Returns true: the file "network_mapping.csv.gz" exists in the directory
+
+// We can create some a new file "network_test.txt.gz" and write "line1" inside
+try (OutputStream os = dataSource.newOutputStream("_test", "txt", false)) {
+    os.write("line1".getBytes(StandardCharsets.UTF_8));
+}
+
+// Another line can be added to the same file by setting the `append` boolean parameter to true
+try (OutputStream os = dataSource.newOutputStream("_test", "txt", true)) {
+    os.write("line2".getBytes(StandardCharsets.UTF_8));
+}
+
+// We can read the file
+try (InputStream is = dataSource.newInputStream("_test", "txt")) {
+    System.out.println(ByteStreams.toByteArray(is)); // Displays "line1" then "line2"
+}
+
+// List the files in the datasource
+Set<String> files = datasource.listNames(".*") // returns a set containing: "network", "network.south", "network.xiidm", "network.v3.xiidm", "network_test.txt"
+```
```

Comment:
More things to eventually document (in this PR or in another):
- Use different datasource on the same directory to select different network
- exists(filename) on a file with a different basename that returns true